presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -535,13 +543,21 @@ private void addToHiveSplitSourceRoundRobin(List<Iterator<HiveSplit>> iteratorLi
             Map<Integer, HiveType> columnCoercions)
             throws IOException
     {
+        Path filePath = new Path(path);
```

Comment:
The code added in this method is to perform a flexible splitting, not for decompression. If the index file is not present, line 592 will make it return the whole file as a split.